VueJS/components/ListItem.js

```diff
@@ -1 +1,60 @@
-export default Vue.component('list-item', {})
+/**
+ * List item component
+ */
+export default Vue.component('list-item', {
+    props: {
+        id: String,
+        message: String,
+    },
+    data: {
+        oldMessage: String
+    },
+    /**
+     * Sets the message value for resetting at cancel.
+     */
+    created: function () {
+        this.oldMessage = this.message
+    },
+    data: function () {
+        return {
+            isHidden: true
+        }
+    },
+    methods: {
+        /**
+         * Toggles the edit form.
+         */
+        showEditForm() {
+            this.isHidden = !this.isHidden
+        },
+        /**
+         * Sets the reset message value when user opens the form.
+         */
+        setOldMessage() {
+            this.oldMessage = this.message
+        },
+        /**
+         * Resets the message value to previously saved value.
+         */
+        resetMessage() {
+            this.$emit('update:message', this.oldMessage)
```

Comment:
```suggestion
            this.$emit('update-message', this.oldMessage)
```
Event names should be kebab-cased: https://vuejs.org/v2/guide/components-custom-events.html